addOns/grpc/src/test/java/org/zaproxy/addon/grpc/internal/VariantGrpcUnitTest.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2024 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.grpc.internal;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.parosproxy.paros.core.scanner.NameValuePair;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+
+public class VariantGrpcUnitTest {
+    private VariantGrpc variantGrpc;
+
+    @BeforeEach
+    void setUp() {
+        variantGrpc = new VariantGrpc();
+    }
+
+    @Test
+    void shouldSetParameterForNestedMessage() throws HttpMalformedHeaderException {
+        String encodedRequestBody =
+                "AAAAAEEKEEhlbGxvLCBQcm90b2J1ZiESJwoESm9obhIGTWlsbGVyGhcKBEpvaG4QAhoNCgtIZWxsbyBXb3JsZBjqrcDlJA";
```

Comment:
Does this contain new lines `\n` like the expected output? 

If not then adding a test in which the input does would be a good move.

Also if permitted by the protocol testing with input payloads that contain multiline strings would be smart too. (Ex `\r\n` and `\n` between the quotes.

Reply:
this payload will give InvalidProtoBufException exception

Reply:
@amitpanwar789 which of those special characters will not give an exception? Can you just test those?

Reply:
Was that because of the colon or something else?

Reply:
Was that because of the colon or something else?

Reply:
because '\n' this will make empty line or Smith will start from new line without and fieldnumber and wiretype and in my encoding algorithm, every key value pair must start with fieldnumber and wiretype